index.bs

```diff
@@ -297,32 +297,40 @@ mechanism for metadata discovery.
 =======================================================
 
-If a listening agent wants to connect to an advertising agent, or to
-learn further metadata about it, it initiates a [[QUIC]] connection to
-the IP and port from the SRV record.  Prior to authentication, a message may be
-exchanged (such as further metadata), but such info should be treated as
-unverified (such as indicating to a user that a display name of an
-unauthenticated agent is unverified).
+If a listening agent ("listener") wants to connect to an advertising agent
+("advertiser"), or to learn further metadata about it, it initiates a [[QUIC]]
+connection to the IP and port from the SRV record.  Prior to authentication,
+a message may be exchanged (such as further metadata), but such info should be
+treated as unverified (such as indicating to a user that a display name of an
+unauthenticated agent is unverified). Note that both listeners and advertisers
+may request information about each other (performing as a "requester" to
+query a "requestee").
 
-To learn further metadata, an agent may send an agent-info-request
-message (see [[#appendix-a]]) and receive back an agent-info-response message.  The
-messages may contain the following information with the following meaning:
+To learn further metadata, devices may send an agent-info-request
+message (see [[#appendix-a]]) and receive back an agent-info-response message.
+Note that both the listener and advertiser may send this request to learn about
+the capabilities of the other device.
+
+The messages may contain the following properties:
 
 : display-name (required)
-:: The display name of the responding agent intended
-    to be displayed to a user by the requesting agent.  If the responding agent
-    is not yet authenticated, the requesting agent should make UI affordance for
-    indicating to the user that the display name is not yet verified.  If the
-    responding agent changes its display name, the requesting agent should
-    make UI affordance for indicating to the user that the display name has
-    changed.
+:: The display name of the requestee, intended to be displayed to a user by the
+     requester. The requester should indicate through the UI if the device
+     is not authenticated or if the display name changes.
 
 : model-name (optional)
-:: If the agent is a hardware device, the model name of
+:: If the requestee is a hardware device, the model name of
     the device.  This is used mainly for debugging purposes, but may be
     displayed to the user of the requesting agent.
 
-<!-- TODO: Add device type and/or capabilities -->
+: supports-audio (optional)
+:: The requestee has to indicate that it supports audio streaming. If false
+    or not included, it is assumed audio streaming is not supported.
+
+: supports-video (optional)
+:: The requestee has to indicate that it supports video streaming. If false
+    or not included, it is assumed video streaming is not supported.
+
```

Comment:
In the future we should consider capabilities for audio input and video input as well.  But these capabilities are more often associated with removable peripherals, so it may not be suitable for device-level metadata.

Maybe we should make all capabilities playback-level metadata?  That is, part of the url-availability-request, media-request and/or remote-playback-availability-request?

Reply:
That's a good question. I'll leave this for now, I'll be in Fremont tomorrow/Friday if you want to sync up in person and hammer down some of the specifics around this.

Reply:
This is an even bigger can of worms, but I'd like to see input device support as a capability too. The trendy new big touchscreen collaborative displays come to mind as the sort of thing one would expect to openscreenprotocol into, which would have their own input device systems.

I don't know whether this capability ought be broken down into smaller pieces- `supports-input` or `supports-touch` `supports-keyboard` `supports-mouse`?

As per previous comments, maybe doesn't need to be done now? I can open a follow-up item depending on peoples feels here.

Reply:
I would vote for a follow up item, personally. I generally prefer smaller patches when possible.